### PR TITLE
Specify default visibility in unit tests

### PR DESCRIPTION
**Pull request by erak**

### CODE REVIEW

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -9932,10 +9932,10 @@ BOOST_AUTO_TEST_CASE(packed_functions)
 	char const* sourceCode = R"(
 		contract C {
 			// these should take the same slot
-			function() external returns (uint) a;
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp]: Why is this change needed?

Comment (erak) [reply test/libsolidity/SolidityEndToEndTest.cpp]: Compiling the contract would fail because ``g``'s type is not implicitly convertible to ``function() external returns (uint)``. This can be fixed by using ``d  = this.g`` and ``a = this.g``, but I still do not fully understand why. Could you quickly elaborate on this?

Comment (erak) [reply test/libsolidity/SolidityEndToEndTest.cpp]: Ok, now I understand why adding ``this`` works. So we need to either leave the contract as it is, or change the visibility of ``a`` and ``d`` to ``external`` and then assign by using ``this.g``.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 2]
```diff
@@ -1800,8 +1800,8 @@ BOOST_AUTO_TEST_CASE(send_ether)
 {
 	char const* sourceCode = R"(
 		contract test {
-			constructor() payable {}
-			function a(address addr, uint amount) returns (uint ret) {
+			constructor() payable public {}
```
Comment (ekpyron) [inline test/libsolidity/SolidityEndToEndTest.cpp:1803]: Not really important, but I think we said visibility first and then mutability :)?

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 3]
```diff
@@ -2545,23 +2545,23 @@ BOOST_AUTO_TEST_CASE(gas_and_value_basic)
 	char const* sourceCode = R"(
 		contract helper {
 			bool flag;
-			function getBalance() payable returns (uint256 myBalance) {
+			function getBalance() payable public returns (uint256 myBalance) {
```
Comment (ekpyron) [inline test/libsolidity/SolidityEndToEndTest.cpp:2548]: Same here.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 4]
```diff
@@ -2578,13 +2578,13 @@ BOOST_AUTO_TEST_CASE(value_complex)
 {
 	char const* sourceCode = R"(
 		contract helper {
-			function getBalance() payable returns (uint256 myBalance) {
+			function getBalance() payable public returns (uint256 myBalance) {
```
Comment (ekpyron) [inline test/libsolidity/SolidityEndToEndTest.cpp:2581]: Same here.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 5]
```diff
@@ -2599,14 +2599,14 @@ BOOST_AUTO_TEST_CASE(value_insane)
...
```
Comment (ekpyron) [inline test/libsolidity/SolidityEndToEndTest.cpp:2602]: Same here.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 6]
```diff
@@ -9479,10 +9479,10 @@ BOOST_AUTO_TEST_CASE(payable_function_calls_library)
 {
 	char const* sourceCode = R"(
 		library L {
-			function f() returns (uint) { return 7; }
+			function f() public returns (uint) { return 7; }
 		}
 		contract C {
-			function f() payable returns (uint) {
+			function f() payable public returns (uint) {
```
Comment (ekpyron) [inline test/libsolidity/SolidityEndToEndTest.cpp:9485]: There's one more!

**Review by ekpyron — COMMENTED** [review]
My vision starts to get blurry after a few-hundred ``public``s :), but I think apart from the few minor comments this should be fine now!

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
If desired, the order problems can be dealt with in a subsequent PR.

File: test/libsolidity/Assembly.cpp [hunk 7]
```diff
@@ -160,19 +160,19 @@ BOOST_AUTO_TEST_CASE(location_test)
 	AssemblyItems items = compileContract(sourceCode);
 	bool hasShifts = dev::test::Options::get().evmVersion().hasBitwiseShifting();
 	vector<SourceLocation> locations =
-		vector<SourceLocation>(hasShifts ? 23 : 24, SourceLocation(2, 75, make_shared<string>(""))) +
-		vector<SourceLocation>(2, SourceLocation(20, 72, make_shared<string>(""))) +
+		vector<SourceLocation>(hasShifts ? 23 : 24, SourceLocation(2, 82, make_shared<string>(""))) +
```
Comment (axic) [inline test/libsolidity/Assembly.cpp:163]: This commit should have been grouped with the code change for this file. Was reviewing this commit by commit and had no idea what this is fixing.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 8]
```diff
@@ -9932,7 +9932,7 @@ BOOST_AUTO_TEST_CASE(packed_functions)
 	char const* sourceCode = R"(
 		contract C {
 			// these should take the same slot
-			function() public returns (uint) a;
```
Comment (axic) [inline test/libsolidity/SolidityEndToEndTest.cpp]: This seems to be a typo and should have been squashed in the original commit :)

**Comment by axic** [comment]
I think this PR should have been squashed prior to merge because it is rather confusing when looking through commits (the same thing is changed 3 times in some cases).
